MediaGalleryUi/Model/SaveImageDetails.php

```diff
@@ -0,0 +1,179 @@
+     * @param LoggerInterface|null $logger
+     */
+    public function __construct(
+        Filesystem $fileSystem,
+        AssetInterfaceFactory $assetFactory,
+        AssetKeywordsInterfaceFactory $assetKeywordsFactory,
+        SaveAssetsInterface $saveAssets,
+        SaveAssetsKeywordsInterface $saveAssetKeywords,
+        AddMetadataInterface $addMetadata,
+        MetadataInterfaceFactory $metadataFactory,
+        LoggerInterface $logger
+    ) {
+        $this->fileSystem = $fileSystem;
+        $this->assetFactory = $assetFactory;
+        $this->assetKeywordsFactory = $assetKeywordsFactory;
+        $this->saveAssets = $saveAssets;
+        $this->saveAssetKeywords = $saveAssetKeywords;
+        $this->addMetadata = $addMetadata;
+        $this->metadataFactory = $metadataFactory;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Save image details
+     *
+     * @param object $asset
+     * @param null|string $imageId
+     * @param array $imageKeywords
+     * @param null|string $imageTitle
+     * @param null|string $imageDescription
+     * @throws LocalizedException
+     */
+    public function execute($asset, $imageId, $imageKeywords, $imageTitle, $imageDescription): void
+    {
+        $updatedAsset = $this->assetFactory->create(
+            [
+                'path' => $asset->getPath(),
+                'contentType' => $asset->getContentType(),
+                'width' => $asset->getWidth(),
+                'height' => $asset->getHeight(),
+                'size' => $asset->getSize(),
+                'id' => $asset->getId(),
+                'title' => $imageTitle,
+                'description' => $imageDescription,
+                'source' => $asset->getSource(),
+                'hash' => $asset->getHash(),
+                'created_at' => $asset->getCreatedAt(),
+                'updated_at' => $asset->getUpdatedAt()
+            ]
+        );
+
+        $this->updateMetadata(
+            $asset->getPath(),
+            $imageTitle,
+            $imageDescription,
+            $imageKeywords
+        );
+
+        try {
+            $this->saveAssets->execute([$updatedAsset]);
+        } catch (CouldNotSaveException $e) {
+            throw new LocalizedException(__($e->getMessage()), $e);
+        }
+
+        $assetKeywords = $this->assetKeywordsFactory->create([
+            'assetId' => $imageId,
+            'keywords' => $imageKeywords
+        ]);
+
+        try {
+            $this->saveAssetKeywords->execute([$assetKeywords]);
+        } catch (CouldNotSaveException $e) {
+            throw new LocalizedException(__($e->getMessage()), $e);
+        }
+    }
+
+    /**
+     * Save updated metadata
+     *
+     * @param object $imagePath
+     * @param null|string $imageTitle
+     * @param null|string $imageDescription
+     * @param array $imageKeywords
+     * @throws FileSystemException
+     */
+    public function updateMetadata(
```

Comment:
Public access is not necessary
```suggestion
    private function updateMetadata(
```